stackit/internal/services/dns/recordset/resource.go

```diff
@@ -417,15 +417,25 @@ func mapFields(recordSetResp *dns.RecordSetResponse, model *Model) error {
 	if recordSet.Records == nil {
 		model.Records = types.ListNull(types.StringType)
 	} else {
-		records := []attr.Value{}
+		respRecords := []string{}
+
 		for _, record := range *recordSet.Records {
-			records = append(records, types.StringPointerValue(record.Content))
+			respRecords = append(respRecords, *record.Content)
+		}
+
+		modelRecords, err := utils.ListValuetoStringSlice(model.Records)
+		if err != nil {
+			return err
 		}
-		recordsList, diags := types.ListValue(types.StringType, records)
+
+		reconciledRecords := utils.ReconcileStringLists(modelRecords, respRecords)
+
+		recordsTF, diags := types.ListValueFrom(context.Background(), types.StringType, reconciledRecords)
```

Comment:
we should use the context coming from the caller method, instead of creating a new one. Pass it as an argument to `mapFields`